### PR TITLE
tools API

### PR DESCRIPTION
**Pull request by ahigerd**

### CODE REVIEW

File: dreamline.pro
Comment (drazil100) [inline dreamline.pro:1]: Not sold on the tool filenames. Like with PolygonItem EdgeItem and GripItem I would like a consistent naming scheme for tools. E.g. VertexTool ColorTool MeshTool etc.

Comment (ahigerd) [reply dreamline.pro:1]: The classes are named that, I just didn't repeat "tool" in both the folder name and file name. I don't have a strong opinion on that and don't mind renaming if you think `src/tools/colortool.h` is better.

Comment (drazil100) [reply dreamline.pro:1]: Oh I see it's in a folder named tools. Gotcha. That said still not sold on the "move" part.

Comment (drazil100) [reply dreamline.pro:1]: Don't know that it matters too much though.

Comment (ahigerd) [reply dreamline.pro:1]: Oh, THAT is ENTIRELY tentative. I don't even know what set of tools we're going to have in the end. These are mostly just sample stubs to show how it's done.

**Pull request merged**